### PR TITLE
Introduce automated testing with GitHub Actions

### PR DESCRIPTION
**Pull request by adrienbernede**
This PR uses GitHub Actions to build GLVis.

This is actually only building GLVis in several configurations, not testing it.

Relates to:
https://github.com/mfem/mfem-actions
mfem/mfem#2146
https://github.com/CEED/Remhos/pull/30
CEED/Laghos#143

### CODE REVIEW

**Comment by adrienbernede** [comment]
Note: I had to use `libfreetype6-dev` for ubuntu-18.04. Apparently we will be able to use `libfreetype-dev` again with ubuntu-20.04...

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      uses: actions/checkout@v2
+      with:
+        path: glvis
+
+    - name: build GLVis
+      run: |
+        glvis_target=""
+        [[ ${{ matrix.target }} == "optim" ]] && glvis_target="opt";
+        [[ ${{ matrix.target }} == "debug" ]] && glvis_target="debug";
+        cd glvis && make ${glvis_target} -j
```
Comment (kanye-quest) [inline .github/workflows/builds.yml:154]: Would it be good to also include a build test for the CMake build system?

Comment (tzanio) [reply .github/workflows/builds.yml:154]: I think it will definitely be good to add that

Comment (adrienbernede) [reply .github/workflows/builds.yml:154]: @kanye-quest does building GLVis with CMake requires building MFEM with CMake first?
I see in the documentation that the example points to MFEMConfig.cmake.

**Comment by adrienbernede** [comment]
I applied the suggested change of name for the action repo (https://github.com/mfem/mfem/pull/2146#issuecomment-816809576).
This uncovered a small bug, that was fixed in the action repo. Now waiting for tests to pass.

File: .github/workflows/builds.yml (hunk 1 above)
Comment (v-dobrev) [reply .github/workflows/builds.yml:154]: Yes, GLVis' CMake build expects MFEM build with CMake.

Comment (adrienbernede) [reply .github/workflows/builds.yml]: Done

**Comment by adrienbernede** [comment]
Looking good

**Review by tzanio — APPROVED** [review]

**Review by kanye-quest — APPROVED** [review]

**Comment by v-dobrev** [comment]
I see that the tests here are using a parallel mfem build. The default should be to use a serial mfem build. Testing against a parallel mfem build is also good to have, but that is an extra.

**Comment by adrienbernede** [comment]
@v-dobrev could you please list the configurations you would like?
We have:
- ubuntu / debug / make
- ubuntu / optim / make 
- macos / debug / make
- macos / optim / make
- ubuntu / release / cmake
All with parallel mfem.
I could move all of them to serial mfem and add "ubuntu / optim / make / mpi ", would that be OK?

**Comment by v-dobrev** [comment]
Since we are just building the executable but not running it, I'm not sure there's need to do both `debug` and `optim`. It is probably good to test with `mfem+mpi` on both `ubuntu` and `macos`. Overall, these are probably good enough:
* ubuntu / optim / make / mfem~mpi
* ubuntu / debug / make / mfem+mpi
* macos / optim / make / mfem~mpi
* macos / debug / make / mfem+mpi
* ubuntu / release / cmake / mfem~mpi

By the way, this is not urgent at all. We can update this later, if everyone agrees it is worth doing at some point.

**Pull request merged**